### PR TITLE
Resolve Issue #2: As a user, I want to be able to navigate between the “list” view and the “add an item” view so that the interface is clearly delinteated between viewing and adding.

### LINKED ISSUE #2: 2. As a user, I want to be able to navigate between the “list” view and the “add an item” view so that the interface is clearly delinteated between viewing and adding.
This is 1 of 2 stories that should be completed in the first week to enable further development of the app.

AC:

* `react-router-dom` has been added as a project dependency
* Links are present and persistent at the bottom of the app: one for the “list” view, the other for the “add an item” view
* When one of the links is clicked, the browser URL updates to represent the current view and a matching view component is displayed
* Whichever view is selected, the corresponding link should display in bold text

### PR DESCRIPTION
**Pull request by thekidnamedkd**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

This code will add react-router-dom to the dependencies. It will then import certain usable components to App.js. With those it will create a simple navigation between "List View" and "Add Item" which will be reflected in the URL path. It will also display the current link selection in bold below per the AC.

## Related Issue

closes #2  

## Acceptance Criteria

* ```react-router-dom``` has been added as a project dependency
* Links are present and persistent at the bottom of the app: one for the “list” view, the other for the “add an item” view
* When one of the links is clicked, the browser URL updates to represent the current view and a matching view component is displayed
* Whichever view is selected, the corresponding link should display in bold text

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### Before

N/A - default app view

### After

after-screenshot

## Testing Steps / QA Criteria

* click links
* check for bold text display
* check that path matches selected link/text display

### CODE REVIEW

**Review by jessicasalbert — APPROVED** [review]

**Review by Maeesha-Rahman — APPROVED** [review]

File: src/App.js [hunk 1]
```diff
@@ -1,26 +1,69 @@
...
+              <AddItem />
+            </Route>
+          </Switch>
+        </header>
+      </div>
+    </Router>
+  );
+}
+
+function MenuLink({ label, to, activeWhenExact }) {
```
Comment (skylerwshaw) [inline src/App.js]: I'd recommend setting a default for `activeWhenExact` here so the `useRouteMatch` hook doesn't receive undefined when the optional param isn't sent. e.g. `activeWhenExact = false`

File: src/App.js [hunk 2]
```diff
@@ -1,26 +1,69 @@
...
+          <MenuLink to="/AddItem" label="Add Item" />
+
+          <Switch>
+            <Route exact path="/ListView">
+              <ListView />
+            </Route>
+            <Route exact path="/AddItem">
+              <AddItem />
+            </Route>
+          </Switch>
```
Comment (skylerwshaw) [inline src/App.js]: Not a need for this PR but we may want to consider when a user lands at a route that doesn't match e.g. `http://localhost:3000/SomeOtherView`. https://reactrouter.com/web/example/no-match

**Review by skylerwshaw — CHANGES_REQUESTED** [review]
Though we could reasonably assume `react-router-dom` API will handle sending `exact: undefined`, we should protect against this case. Aside from this small fixup I think the PR is 👌 great.

**Comment by esolar07** [comment]
@thekidnamedkd You might need to pull from `main` so you can updated your branch before we merge it

File: src/App.js (hunk 1 above)
Comment (skylerwshaw) [reply src/App.js]:
```suggestion
function MenuLink({ label, to, activeWhenExact = false }) {
```

I've included a code change you can accept into a commit here for what I was referring to in the comment above.

**Review by skylerwshaw — APPROVED** [review]

**Pull request merged**